app/src/pages/Peers.tsx

```diff
@@ -12,30 +13,46 @@ import {
   GradientHeading,
   PageHead,
   PageWrapper,
-  PeerTable,
+  PeersTable,
 } from '../components';
 
 export const Peers: React.FC = () => {
+  const [isTableLoading, setIsTableLoading] = useState(false);
+
   const { t } = useTranslation();
 
   const dispatch = useAppDispatch();
 
-  useEffect(() => {
-    dispatch(fetchPeers());
-  }, [dispatch]);
-
   const peers = useAppSelector(getPeers);
+  const peersTableOptions = useAppSelector(getPeersTableOptions);
   const peersLoadingStatus = useAppSelector(getPeerLoadingStatus);
 
-  const isLoading = peersLoadingStatus !== Loading.Complete;
+  useEffect(() => {
+    dispatch(fetchPeers(peersTableOptions));
+  }, [dispatch, peersTableOptions]);
+
+  useEffect(() => {
+    if (isTableLoading) {
+      setIsTableLoading(false);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [peers]);
+
+  const isLoading = peersLoadingStatus !== Loading.Complete && !peers.length;
 
   const pageTitle = `${t('peers')}`;
 
   return (
     <PageWrapper isLoading={isLoading}>
       <PageHead pageTitle={pageTitle} />
       <GradientHeading type="h2">{t('connected-peers')}</GradientHeading>
-      {peers && <PeerTable peers={peers} />}
+      {peers && (
```

Comment:
We shouldn't be conditionally rendering this table. All logic to show data or not should be handled inside the PeersTable. See how I did it in the blocks or validators tables. We need to have the table loaded immediately with a loading indicator showing that we are fetching the table content - again, much like is done in the blocks/validators table.